src/service/resources-impl.js

```diff
@@ -1721,7 +1729,17 @@ export class Resource {
    * @return {boolean}
    */
   renderOutsideViewport() {
-    return this.element.renderOutsideViewport();
+    const renders = this.element.renderOutsideViewport();
+    // Boolean interface, element is either always allowed or never allowed to
+    // render outside viewport.
+    if (renders === true || renders === false) {
+      return renders;
+    }
+    // Numeric interface, element is allowed to render outside viewport when it
+    // is within X times the viewport height of the current viewport.
+    const viewportBox = this.resources_.getViewport().getRect();
+    const distanceFromViewport = this.layoutBox_.top - viewportBox.bottom;
```

Comment:
Hmm. This is not how `renderOutsideViewport` used to work. This affects performance directly since the user would have to wait to see for what's in the viewport right now longer. I think we should change to `abs` or a above/below formula. Seems like the latter is better since you'd have to do `this.layoutBox_.bottom - viewportBox.top` instead for the above case.